fmlcore/src/main/java/net/minecraftforge/fml/ModLoader.java

```diff
@@ -184,21 +184,27 @@ public void gatherAndInitializeMods(final ModWorkManager.DrivenExecutor syncExec
         this.modList = modList;
         var stateList = stateManager.getStates(ModLoadingPhase.GATHER);
         var progress = StartupMessageManager.addProgressBar("Mod Gather", stateList.stream().mapToInt(mls -> mls.size().applyAsInt(modList)).sum());
-        stateList.forEach(mls->dispatchAndHandleError(mls, syncExecutor, parallelExecutor, periodicTask, progress));
+        for (IModLoadingState mls : stateList) {
```

Comment:
Feel free to unfold all of the lambdas. As 1) sometimes they are faster and 2) they are always easier to debug.
The line above this is a prime example as it's calculating a sum using streams and tons of indirect functions.

Reply:
I was referring to `stateList.stream().mapToInt(mls -> mls.size().applyAsInt(modList)).sum()`
And as for your example of the sizing issues, you can minimize that by `new ArrayList<>(this.loadingExceptions.size())` but yes thats always the balance between things. Lambdas are more compact, but less understandable. 

Basically what I am saying, is that we are both agreeing that streams are overused, and if you feel like it de-streamifying things is something I'm encouraging you to do. Besides that I've already said the rest of the cleanups in this PR are fine.